db/migrate/20160414094300_change_capacity_to_hash_from_persistent_volume.rb

```diff
@@ -0,0 +1,56 @@
+class ChangeCapacityToHashFromPersistentVolume < ActiveRecord::Migration[5.0]
+  class ContainerVolume < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+  end
+
+  class PersistentVolume < ContainerVolume
+    self.inheritance_column = :_type_disabled # disable STI
+    serialize :capacity
+
+    IEC_SIZE_SUFFIXES = %w(Ki Mi Gi Ti).freeze
+    def self.parse_iec_number(value)
+      exp_index = IEC_SIZE_SUFFIXES.index(value[-2..-1])
+      if exp_index.nil?
+        return Integer(value)
+      else
+        return Integer(value[0..-3]) * 1024**(exp_index + 1)
+      end
+    end
+  end
+
+  def up
+    change_column :container_volumes, :capacity, :text
+    say_with_time("Changing string to hash") do
+      PersistentVolume.all.each do |vol|
+        result_hash = {}
+        vol.capacity.split(',').each do |hash|
+          key, val = hash.split('=')
+          next if val.nil?
+          begin
+            result_hash[key.to_sym] =  PersistentVolume.parse_iec_number val
+          rescue
+            result_hash[key.to_sym] = val
+          end
+        end
+        vol.update_attributes!(:capacity => result_hash)
+      end
+    end
+  end
+
+  def down
+    say_with_time("Changing hash to string") do
+      PersistentVolume.all.each do |vol|
+        result_string = nil
+        unless vol.capacity.empty?
+          result_string = ""
+          vol.capacity.each do |key, val|
+            result_string += "#{key}=#{val},"
```

Comment:
Prefer Array#join (which would also prevent the chop! at the end).  You can also avoid the various initializations of result_string, by leveraging the fact that collect on an empty collection returns an empty array.

```ruby
PersistentVolume.all.each do |vol|
  capacity = vol.capacity.collect { |key, val| "#{key}=#{val}" }.join(",")
  capacity = nil if capacity.blank?
  vol.update_attributes!(:capacity => capacity)
end